File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Rules/OrderProcessingPolicy/VB/OrderProcessingPolicy/Workflow.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Drawing
Imports System.Text
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.Runtime

Namespace Microsoft.Rules.Samples
    Partial Public NotInheritable Class Workflow
        Inherits SequentialWorkflowActivity
        Private m_customerName As String ' ie. "John Customer"
        Private m_itemNum As Integer ' ie. 1 => for Vista Ultimate DVD
        Private m_zipCode As String ' ie. "00999"
        Private invalidOrder As OrderError
        Private invalidZipCodeErrorCollection As OrderErrorCollection
        Private invalidItemNumErrorCollection As OrderErrorCollection
        Private invalidOrdersCollection As OrderErrorCollection

        Public Property CustomerName() As String
            Get
                Return Me.m_customerName
            End Get
            Set(ByVal value As String)
                Me.m_customerName = Value
            End Set
        End Property

        Public Property ItemNum() As Integer
            Get
                Return Me.m_itemNum
            End Get
            Set(ByVal value As Integer)
                Me.m_itemNum = value
            End Set
        End Property

        Public Property ZipCode() As String
            Get
                Return Me.m_zipCode
            End Get
            Set(ByVal value As String)
                Me.m_zipCode = Value
            End Set
        End Property

        Public Sub New()
            InitializeComponent()
        End Sub
    End Class
End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Rules/OrderProcessingPolicy/CS/OrderProcessingPolicy/Workflow.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Drawing;
using System.Text;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;

#pragma warning disable 169     // Ignore fields not used because they are used by Rules.

namespace Microsoft.Rules.Samples
{
    public sealed partial class Workflow : SequentialWorkflowActivity
    {
        private string customerName;                // that is,. "John Customer".
        private int itemNum;                        // that is, 1 => for Vista Ultimate DVD.
        private string zipCode;                     // that is, "00999".
        private OrderError invalidOrder;
        private OrderErrorCollection invalidZipCodeErrorCollection;
        private OrderErrorCollection invalidItemNumErrorCollection;
        private OrderErrorCollection invalidOrdersCollection;

        public string CustomerName
        {
            get
            {
                return this.customerName;
            }
            set
            {
                this.customerName = value;
            }
        }

        public int ItemNum
        {
            get
            {
                return this.itemNum;
            }
            set
            {
                this.itemNum = value;
            }
        }

        public string ZipCode
        {
            get
            {
                return this.zipCode;
            }
            set
            {
                this.zipCode = value;
            }
        }

        public Workflow()
        {
            InitializeComponent();
        }
    }
}
